src/db_state.rs

```diff
@@ -399,6 +408,19 @@ impl DbState {
         self.update_state(state);
     }
 
+    /// increment_seq is called whenever a new write is performed.
+    pub fn increment_seq(&mut self) -> u64 {
+        let last_seq = self.last_seq;
```

Comment:
Curious why not:

```rust
self.last_seq += 1;
self.last_seq
```

Are you trying to avoid a read-modify-write race condition? Wouldn't `+=` still have that issue?

Reply:
I had the same thought. This is always accessed using a write guard, so there shouldn't be a race here.

Reply:
yeah, there's a Mutex outside of DbState

Reply:
@flaneur2020 did you want to change it then? Or was there another reason?

Reply:
sorry, i only noticed the discussion around possible race condition earlier. yes, this code should indeed be modified. fixed in d50a075